src/providers/kick.ts

```diff
@@ -1,43 +1,72 @@
-
+import { createOAuth2Request, sendTokenRequest, sendTokenRevocationRequest } from "../request.js";
 import type { OAuth2Tokens } from "../oauth2.js";
 
 const authorizationEndpoint = "https://id.kick.com/oauth/authorize";
 const tokenEndpoint = "https://id.kick.com/oauth/token";
 const tokenRevocationEndpoint = "https://id.kick.com/oauth/revoke";
 
 export class Kick {
-	private client: OAuth2Client;
+	private clientId: string;
+	private clientSecret: string;
+	private redirectURI: string;
 
-	constructor(clientId: string, clientSecret: string | null, redirectURI: string) {
-		this.client = new OAuth2Client(clientId, clientSecret, redirectURI);
+	constructor(clientId: string, clientSecret: string, redirectURI: string) {
+		this.clientId = clientId;
+		this.clientSecret = clientSecret;
+		this.redirectURI = redirectURI;
 	}
 
 	public createAuthorizationURL(state: string, codeVerifier: string, scopes: string[]): URL {
-		const url = this.client.createAuthorizationURLWithPKCE(
-			authorizationEndpoint,
-			state,
-			CodeChallengeMethod.S256,
-			codeVerifier,
-			scopes
-		);
+		const url = new URL(authorizationEndpoint);
+		url.searchParams.set("client_id", this.clientId);
+		url.searchParams.set("response_type", "code");
+		url.searchParams.set("redirect_uri", this.redirectURI);
+		url.searchParams.set("state", state);
+		if (scopes.length > 0) {
+			url.searchParams.set("scope", scopes.join(" "));
+		}
+		url.searchParams.set("code_challenge", codeVerifier);
+		url.searchParams.set("code_challenge_method", "S256");
 		return url;
 	}
 
 	public async validateAuthorizationCode(
 		code: string,
 		codeVerifier: string
 	): Promise<OAuth2Tokens> {
-		const tokens = await this.client.validateAuthorizationCode(tokenEndpoint, code, codeVerifier);
+		const body = new URLSearchParams();
+		body.set("code", code);
+		body.set("client_id", this.clientId);
+		body.set("client_secret", this.clientSecret);
+		body.set("redirect_uri", this.redirectURI);
+		body.set("grant_type", "authorization_code");
+		body.set("code_verifier", codeVerifier);
+		const request = createOAuth2Request(tokenEndpoint, body);
+		const tokens = await sendTokenRequest(request);
 		return tokens;
 	}
 
 	public async refreshAccessToken(refreshToken: string): Promise<OAuth2Tokens> {
-		const tokens = await this.client.refreshAccessToken(tokenEndpoint, refreshToken, []);
+		const body = new URLSearchParams();
+		body.set("refresh_token", refreshToken);
+		body.set("client_id", this.clientId);
+		body.set("client_secret", this.clientSecret);
+		body.set("grant_type", "refresh_token");
+		const request = createOAuth2Request(tokenEndpoint, body);
+		const tokens = await sendTokenRequest(request);
 		return tokens;
 	}
 
-	public async revokeToken(token: string): Promise<void> {
-		await this.client.revokeToken(tokenRevocationEndpoint, token);
+	public async revokeToken(
+		token: string,
+		tokenHintType?: "access_token" | "refresh_token"
```

Comment:
The spec actually requires the provider to check for both token type regardless of the `token_type_hint` parameter value. So you can just remove it (all other providers here don't have it either)